File: PharmaCity.Angular/src/models/PurchaseDTO.ts
import { Petition } from "./Petition";
import { Pharmacy } from "./Pharmacy";

export class PurchaseDTO{

    id: number;
    purchaseCode: string;
    shopping: Array<Petition>;
    state: string;

    constructor(){
        this.id = 0;
        this.purchaseCode = "";
        this.shopping = [];
        this.state = "";
    }
}

File: PharmaCity.System/PharmaCity.Domain/DTO/PurchaseDTO.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace PharmaCity.Domain.DTO
{
    public class PurchaseDTO
    {
        public int Id { get; set; }
        public string PurchaseCode { get; set; }
        public string PharmacyName { get; set; }
        public IEnumerable<PetitionBuy> Shopping { get; set; }
        public string State { get; set; }
    }
}
